ui-trackoccupancydiagram/src/components/helpers/drawElements/drawTracks.ts

```diff
@@ -0,0 +1,115 @@
+import { drawTrack } from './drawTrack';
+import { TRACK_HEIGHT_CONTAINER, CANVAS_PADDING, COLORS, TICKS_PRIORITIES } from '../../consts';
+import { type Track } from '../../types';
+
+export function getTimeToPixel(
+  timeOrigin: number,
+  pixelOffset: number,
+  timeScale: number
+): (time: number) => number {
+  return (time: number) => pixelOffset + (time - timeOrigin) / timeScale;
+}
+
+const { WHITE_100, HOUR_BACKGROUND } = COLORS;
+
+const drawBackground = ({
+  ctx,
+  xStart,
+  width,
+  height,
+  switchBackground,
+}: {
+  ctx: CanvasRenderingContext2D;
+  xStart: number;
+  width: number;
+  height: number;
+  switchBackground: boolean;
+}) => {
+  if (xStart >= 0) {
+    ctx.clearRect(xStart, 0, width, height);
+    ctx.fillStyle = switchBackground ? HOUR_BACKGROUND : WHITE_100;
+    ctx.fillRect(xStart, 0, width, height);
+  }
+};
+
+type DrawTracksProps = {
+  ctx: CanvasRenderingContext2D;
+  width: number;
+  height: number;
+  tracks: Track[] | undefined;
+  timeOrigin: number;
+  timePixelOffset: number;
+  getTimePixel: (time: number) => number;
+  timeRanges: number[];
+  breakpoints: number[];
+  timeScale: number;
+};
+
+export const drawTracks = ({
+  ctx,
+  width,
+  height,
+  tracks,
+  timeOrigin,
+  timePixelOffset,
+  getTimePixel,
+  timeRanges,
+  breakpoints,
+  timeScale,
+}: DrawTracksProps) => {
+  ctx.clearRect(0, 0, width, height);
+
+  const minT = timeOrigin - timeScale * timePixelOffset;
+  const maxT = minT + timeScale * width;
+  const pixelsPerMinute = (1 / timeScale) * 60_000;
+  let labelLevels: number[] = [];
+
+  breakpoints.some((breakpoint, i) => {
+    if (pixelsPerMinute < breakpoint) {
+      labelLevels = TICKS_PRIORITIES[i];
+      return true;
+    }
+    return false;
+  });
+
```

Comment:
same as below you could extract the logic in a dedicated function to improve lisibility

Reply:
done

Reply:
done